src/amo/components/ReportAbuseButton/index.js

```diff
@@ -0,0 +1,220 @@
+import classNames from 'classnames';
+import { oneLine } from 'common-tags';
+import defaultDebounce from 'lodash.debounce';
+import React from 'react';
+import { connect } from 'react-redux';
+import Textarea from 'react-textarea-autosize';
+import { compose } from 'redux';
+
+import { withErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import {
+  disableAbuseButtonUI,
+  enableAbuseButtonUI,
+  hideAddonAbuseReportUI,
+  sendAddonAbuseReport,
+  showAddonAbuseReportUI,
+} from 'core/reducers/abuse';
+import { sanitizeHTML } from 'core/utils';
+import Button from 'ui/components/Button';
+
+import './styles.scss';
+
+
+type PropTypes = {
+  abuseReport: {| message: string, reporter: Object | null |},
+  addon: Object | null,
+  debounce: any,
+  dispatch: any,
+  errorHandler: any,
+  loading: bool,
+  i18n: Object,
+};
+
+export class ReportAbuseButtonBase extends React.Component {
+  static defaultProps = {
+    debounce: defaultDebounce,
+  };
+
+  cancelReport = (event) => {
+    event.preventDefault();
+
+    const { addon, dispatch, loading } = this.props;
+
+    if (loading) {
+      return;
+    }
+
+    dispatch(hideAddonAbuseReportUI({ addon }));
+  }
+
+  sendReport = (event) => {
+    event.preventDefault();
+
+    // The button isn't clickable if there is no content, but because we
+    // debounce the checks we just verify there's a message to send.
+    if (!this.textarea.value.length) {
+      log.debug(oneLine`User managed to click submit button before
+        disableAbuseButtonUI() was called because of debouncing textarea
+        onChange event. Ignoring this onClick event.`);
+      return;
+    }
+
+    const { addon, dispatch, errorHandler } = this.props;
+
+    dispatch(sendAddonAbuseReport({
+      addonSlug: addon.slug,
+      errorHandlerId: errorHandler.id,
+      message: this.textarea.value,
+    }));
+  }
+
+  showMore = (event) => {
+    event.preventDefault();
+
+    const { addon, dispatch } = this.props;
+
+    dispatch(showAddonAbuseReportUI({ addon }));
+    this.textarea.focus();
+  }
+
+  textareaChange = this.props.debounce(() => {
```

Comment:
We actually don't save this content yet though I think we could/should in the future.

I was debouncing because I thought firing it on every change event seemed excessive. But I honestly forget how controlled inputs work–does that mean React is already debouncing these type of events? If that's the case I'll remove this, but I just didn't want the event to be firing on every character entered...

Reply:
Events are DOM/browser events, they are not denounced by react and we usually don't have to do that.